security/libssh:

* Update to 0.8.3, fixes OPENSSL-devel build
* Re-order variables, per handbook

Changes:	https://www.libssh.org/2018/09/21/libssh-0-8-3/

PR:		228895
Submitted by:	Nathan <ndowens@yahoo.com

Index: security/libssh/Makefile
===================================================================
--- security/libssh/Makefile	(revision 481216)
+++ security/libssh/Makefile	(working copy)
@@ -2,9 +2,9 @@
 # $FreeBSD$
 
 PORTNAME=	libssh
-DISTVERSION=	0.7.5
+DISTVERSION=	0.8.3
 CATEGORIES=	security devel
-MASTER_SITES=	https://git.libssh.org/projects/libssh.git/snapshot/
+MASTER_SITES=	https://www.libssh.org/files/${PORTVERSION:R}/
 
 MAINTAINER=	johans@FreeBSD.org
 COMMENT=	Library implementing the SSH1 and SSH2 protocol
@@ -12,15 +12,6 @@
 LICENSE=	LGPL21
 LICENSE_FILE=	${WRKSRC}/COPYING
 
-BROKEN_SSL=	openssl-devel
-BROKEN_SSL_REASON_openssl-devel=	error: variable has incomplete type 'EVP_MD_CTX' (aka 'struct evp_md_ctx_st')
-
-OPTIONS_DEFINE=	SSH1 STATIC
-OPTIONS_RADIO=	CRYPTO
-OPTIONS_RADIO_CRYPTO=	GCRYPT OPENSSL
-OPTIONS_DEFAULT=	OPENSSL STATIC
-OPTIONS_SUB=	yes
-
 USES=		cmake:outsource cpe tar:xz
 USE_LDCONFIG=	yes
 
@@ -29,16 +20,21 @@
 		-DWITH_TESTING:BOOL=OFF \
 		-DWITH_ZLIB:BOOL=ON
 
+OPTIONS_DEFINE=		SFTP STATIC
+OPTIONS_RADIO=		CRYPTO
+OPTIONS_RADIO_CRYPTO=	GCRYPT OPENSSL
+OPTIONS_DEFAULT=	OPENSSL SFTP STATIC
+OPTIONS_SUB=		yes
+
 GCRYPT_LIB_DEPENDS=	libgcrypt.so:security/libgcrypt
 GCRYPT_CMAKE_ON=	-DWITH_GCRYPT:BOOL=ON
 GCRYPT_CMAKE_OFF=	-DWITH_GCRYPT:BOOL=OFF
 
 OPENSSL_USES=		ssl
+SFTP_DESC=		Build with SFTP support
+SFTP_CMAKE_ON=		-DWITH_SFTP:BOOL=ON
+SFTP_CMAKE_OFF=		-DWITH_SFTP:BOOL=OFF
 
-SSH1_DESC=		Build with SSH1 support
-SSH1_CMAKE_ON=		-DWITH_SSH1:BOOL=ON
-SSH1_CMAKE_OFF=		-DWITH_SSH1:BOOL=OFF
-
 STATIC_CMAKE_ON=	-DWITH_STATIC_LIB:BOOL=ON
 STATIC_CMAKE_OFF=	-DWITH_STATIC_LIB:BOOL=OFF
 
@@ -48,8 +44,6 @@
 		-e 's|LIB_INSTALL_DIR}|CMAKE_INSTALL_PREFIX}/libdata|' \
 		-e 's|CMAKE_INSTALL_DIR}|CMAKE_INSTALL_DIR}/libssh|' \
 		${WRKSRC}/CMakeLists.txt
-	@${REINPLACE_CMD} -e 's|-lssh_threads|-lssh_threads -lpthread|' \
-		${WRKSRC}/libssh_threads.pc.cmake
 	@${REINPLACE_CMD} -e 's| -pedantic-errors||' \
 		${WRKSRC}/cmake/Modules/DefineCompilerFlags.cmake
 
Index: security/libssh/distinfo
===================================================================
--- security/libssh/distinfo	(revision 481216)
+++ security/libssh/distinfo	(working copy)
@@ -1,3 +1,3 @@
-TIMESTAMP = 1493578451
-SHA256 (libssh-0.7.5.tar.xz) = 6f7d5b9f3089164eec7408d35b99194b17352cce436a4a95485b511bce2aab4b
-SIZE (libssh-0.7.5.tar.xz) = 351364
+TIMESTAMP = 1538696697
+SHA256 (libssh-0.8.3.tar.xz) = 302f31f606f2368cd3ce77d7a69f7464c18eae176e73e59102e0524401bd29d0
+SIZE (libssh-0.8.3.tar.xz) = 422244
Index: security/libssh/pkg-plist
===================================================================
--- security/libssh/pkg-plist	(revision 481216)
+++ security/libssh/pkg-plist	(working copy)
@@ -3,18 +3,12 @@
 include/libssh/libssh.h
 include/libssh/libsshpp.hpp
 include/libssh/server.h
-include/libssh/sftp.h
-%%SSH1%%include/libssh/ssh1.h
+%%SFTP%%include/libssh/sftp.h
 include/libssh/ssh2.h
 lib/cmake/libssh/libssh/libssh-config.cmake
 lib/cmake/libssh/libssh/libssh-config-version.cmake
 %%STATIC%%lib/libssh.a
+lib/libssh.so.4.7.0
+lib/libssh.so.4
 lib/libssh.so
-lib/libssh.so.4
-lib/libssh.so.4.4.2
-%%STATIC%%lib/libssh_threads.a
-lib/libssh_threads.so
-lib/libssh_threads.so.4
-lib/libssh_threads.so.4.4.2
 libdata/pkgconfig/libssh.pc
-libdata/pkgconfig/libssh_threads.pc
